docs/semgrep-app/notifications.md

```diff
@@ -277,31 +277,144 @@ Automated comments on GitLab merge requests are displayed as follows:
+1. In the Bitbucket workspace, click the cogwheel icon next to your profile icon in the top right.
+1. Click **Workspace settings** > **Workspace variables**.
+1. Create a new workspace variable by entering **SEMGREP_APP_TOKEN** into the **Name** field, and then paste the value of the token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+##### Creating and adding a repository access token
+
+Create a repository access token in Bitbucket. Follow the instructions in [Create a repository Access Token](https://support.atlassian.com/bitbucket-cloud/docs/create-a-repository-access-token/). Alternatively, follow these steps to create a repository access token:
+
+1. In Bitbucket, select the repository where you want Semgrep to run in CI.
+1. Click **Repository settings** > **Access tokens**.
+1. Click **Create Repository Access Token**.
+1. Create a token name. Note: As Semgrep comments to your PRs, use **Semgrep** as the name for this token.
+1. Under **Scopes**, select the **Read**, and **Write** permissions for pull requests.
+1. Click **Create**.
+1. Copy the value of the repository access token, and then click **Close**.
+1. Click **Repository variables**.
+1. Create a new repository variable by entering **PAT** into the **Name** field, and then paste the workspace access token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+Create and add a `SEMGREP_APP_TOKEN` to establish the communication between your repository and the Semgrep Cloud Platform by following these steps:
+
+1. Create `SEMGREP_APP_TOKEN` by following the [Creating a `SEMGREP_APP_TOKEN`](/semgrep-ci/running-semgrep-ci-with-semgrep-app/#creating-a-semgrep_app_token) documentation. Copy the value of the token created in the Semgrep Cloud Platform.
+1. In Bitbucket, select the repository where you want Semgrep to run in CI.
+1. Click **Repository settings** > **Repository variables**.
+1. Create a new repository variable by entering **SEMGREP_APP_TOKEN** into the **Name** field, and then paste the value of the token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+#### Adding Semgrep to your Bitbucket CI pipeline for PR comments
+
+To receive comments on PRs, add the following code to your `bitbucket-pipelines.yml` file:
+
+```yaml
+image: atlassian/default-image:latest
+
+pipelines:
+  pull-requests:
+    '**':
+      - step:
+          name: 'Run Semgrep scan in PR branch'
+          image: returntocorp/semgrep
+          script:
+            - export SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN
+            - export SEMGREP_APP_URL="https://semgrep.dev"
+            - export SEMGREP_BASELINE_REF="origin/main"
+            - git fetch origin "+refs/heads/*:refs/remotes/origin/*"
+            - export BITBUCKET_TOKEN=$PAT
+            - semgrep ci
+```
```

Comment:
@pangngan let me know what we need to edit here to make this scan the main branches also. :) I was looking here - https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-bitbucket-pipelines-configuration-snippet should we update the code snippet too?

Reply:
The reason why the pipeline event trigger is `pull-requests` (line 393) is because you can't generate a PR comment on a MAIN (or default) branch. It's in the name of the comment: PR comment.